### ISSUE TITLE
[Bug]: symmetric_megnitude zeroes the magnitude instead of negating it, so Rotate/Shear/Translate never go negative

### ISSUE THREAD

**Issue by mohansree14** [kornia/kornia#4494]
### What happened?

`OperationBase._init_magnitude_fn` multiplies the sampled magnitude by a **boolean** mask instead of a sign, so `symmetric_megnitude=True` zeroes half the magnitudes rather than negating them.

https://github.com/kornia/kornia/blob/main/kornia/augmentation/auto/operations/base.py#L82-L87

```python
def _random_flip(fn):
    def f(x):
        flip = torch.rand((x.shape[0],), device=x.device) > 0.5
        return fn(x) * flip          # <-- bool, so this is `m` or `0`, never `-m`
    return f
```

`flip` is a `torch.bool` tensor, so `fn(x) * flip` promotes to `m * 1` or `m * 0`. The intended `-m` branch never happens.

Two consequences:

1. **The augmentations are one-sided.** `Rotate`, `ShearX`, `ShearY`, `TranslateX` and `TranslateY` all default to `symmetric_megnitude=True` and are the only ops that use it. Images are only ever rotated clockwise, only ever sheared and translated in the positive direction. That is not AutoAugment/RandAugment/TrivialAugment as specified, and it halves the geometric diversity of every policy that draws one of these five.
2. **Roughly half of all draws are silently a no-op.** A zero magnitude means the operation runs but does nothing, so a policy that picks `rotate` leaves the image unchanged about half the time.

The docstring says the opposite of what the code does — `symmetric_megnitude: if to randomly assign the magnitude as negative or not` (`operations/ops.py:414`) — and the callers are built around that reading: `Rotate.__init__` rejects `magnitude_range[0] < 0` when symmetric, and `rand_augment.ops.rotate` requires `min_mag == -max_mag` then passes `(0.0, max_mag)`. Both only make sense if the sign is applied afterwards.

**Expected:** magnitudes distributed symmetrically about zero, with no spike at zero.

### Minimal reproduction

```python
import torch
from kornia.augmentation.auto.operations import ops

torch.manual_seed(0)
for cls, factor in [(ops.Rotate, "degrees"), (ops.ShearX, "shear_x"), (ops.TranslateX, "translate_x")]:
    op = cls(None, 1.0)  # symmetric_megnitude=True by default
    m = torch.cat([op.forward_parameters(torch.Size([64, 3, 8, 8]))[factor] for _ in range(40)])
    print(f"{cls.__name__:10s} neg={(m < 0).sum().item():4d} zero={(m == 0).sum().item():4d} pos={(m > 0).sum().item():4d}")
```

Actual, 2560 draws each:

```
Rotate     neg=   0 zero=1258 pos=1302     # min=0.000  max=29.975, never below 0
ShearX     neg=   0 zero=1238 pos=1322
TranslateX neg=   0 zero=1259 pos=1301
```

Expected: `neg` and `pos` each near half, `zero` near 0.

### Environment

- Kornia version: 0.9.0rc1 (also present on `main`)
- PyTorch version: 2.14.0+cpu
- Python version: 3.12.7
- OS / device (CPU, CUDA, ROCm, MPS): Windows 11, CPU

### Anything else?

Applying a sign instead of a mask gives the documented distribution:

```python
sign = torch.where(torch.rand((x.shape[0],), device=x.device) > 0.5, 1.0, -1.0).to(x.dtype)
return fn(x) * sign
```

With that change, same script and seed:

```
Rotate     neg=1258 zero=   0 pos=1302     # min=-29.995  max=29.975
ShearX     neg=1238 zero=   0 pos=1322
TranslateX neg=1259 zero=   0 pos=1301
```

The explicit `.to(x.dtype)` matters — `torch.where` on Python floats produces `float32` and would otherwise silently upcast a half-precision magnitude.

Note this is a behaviour change for anyone with a seeded pipeline, since the sampled magnitudes move.

Found while writing the geometry-only inverse round-trip test in #4456 — the zero-magnitude draws made a share of that test's assertions vacuous. Not fixed there, it is unrelated to that PR's guard. Happy to send a PR with a regression test on the magnitude distribution if that is wanted.
